lib/ecto/adapters/postgres/connection.ex

```diff
@@ -789,6 +789,10 @@ if Code.ensure_loaded?(Postgrex) do
       ?t
     end
 
+    defp table_type_expr(nil), do: "TABLE "
+    defp table_type_expr(type) when is_binary(type), do: type <> " "
+    defp table_type_expr(type), do: raise ArgumentError, message: "Table type must be a String, got " <> inspect(type)
```

Comment:
Just for consistency with other error messages:

```suggestion
    defp table_type_expr(type), do: raise ArgumentError, "table type must be a string, got `#{inspect(type)}`"
```